### ISSUE TITLE
TypeError: Cannot read property 'counter' of undefined - on the auto-created subscriptions

### ISSUE THREAD

**Issue by hkeeper** [guidesmiths/rascal#78]
Hallo!

I've created simple config file(see in the end of message) and used it in the code from examples

```
const Broker = require('rascal').BrokerAsPromised;
const config = require('./config');
 (async () => {
  try {
    **const broker = await Broker.create(config);**
    broker.on('error', console.error);
```
But got the error
TypeError: Cannot read property 'counter' of undefined
    at validateSubscription (c:\Src\rascal\node_modules\rascal\lib\config\validate.js:142:65)

It seems that for auto-created subscription,
subscription.redeliveries is empty

So
subscription.redeliveries.counter throws error

![image](https://user-images.githubusercontent.com/28985646/73659261-ee045c00-46a6-11ea-8391-6ad889a60276.png)

**Config.json**
```
{
  "vhosts": {
    "rabbit": {
      "connection": {
        "url": "amqp://usr:pass@app-serv:5742/"
      },
      "queues": [
        "1C"
      ]
    }
  }
}
```

**Comment by cressie176** [comment]
Hi @hkeeper,

Thanks for reporting. I'll look into it as soon as possible. In the meantime you might get somewhere by using...
```js
const decoratedConfig = rascal.withDefaultConfig(config);
await Broker.create(decoratedConfig)
```

**Comment by cressie176** [comment]
I'm going to fix this by moving most of the settings applied through `withDefaultConfig` into a new baseline config, which is always applied. The only settings applied through `withDefaultConfig` from now on will be things that cannot be easily overwritten or removed, e.g. it would be difficult to remove connection attributes such as heartbeat, connection_timeout, channelMax.

**Comment by cressie176** [comment]
Published as rascal@9.3.0
